### ISSUE TITLE
support of MA0002 with HashSet<string> and [] syntax

### ISSUE THREAD

**Issue by flucas1** [meziantou/Meziantou.Analyzer#1239]
This raises MA0002
`HashSet<string> hashesOld = new();`

But this is ignored
`HashSet<string> hashesNew = [];`

I suggest to unify the behaviour and suggest
`HashSet<string> hashesNew = [with(StringComparer.Ordinal)];`

**Comment by meziantou** [comment]
Are you using C# 15 (preview)? If not, you may need to opt-in the behavior:

editorconfig:
```
# Report Dictionary<string, TValue> collection expressions ([]) for C#14 and lower
# C# preview reports those diagnostics by default
MA0002.report_collection_expressions = true
```

**Comment by meziantou** [comment]
I think I've found the issue. While testing yesterday, I special-cased Dictionary. Let's generalize it to all collection expressions.

**Comment by flucas1** [comment]
I use this,

		<TargetFramework>net11.0</TargetFramework>
		<LangVersion>preview</LangVersion>
		<EnablePreviewFeatures>true</EnablePreviewFeatures>
		<AnalysisLevel>preview</AnalysisLevel>
